### ISSUE TITLE
PrivateReader from connection

### ISSUE THREAD

**Issue by joshua-oss** [opendp/smartnoise-sdk#377]
Readers can now be instantiated from a connection.  The PrivateReader should allow instantiation from a connection, with no reader explicitly created.

```
conn = pyodbc.connect(dsn)
private_reader = PrivateReader(conn, meta, 1.0)
```

This calling pattern overloads the `reader` parameter on the `PrivateReader` constructor, passing in a connection rather than a reader.  This would work fine, but we already have overloaded the first two parameters of the constructor.  A better approach might be to use the factory pattern used by `SqlReader`, like:

```
conn = pyodbc.connect(dsn)
privacy = Privacy(epsilon=1.0, delta=10E-16)
private_reader = PrivateReader.from_connection(conn, privacy, meta=meta)
```

This has the advantage that it matches the preferred calling pattern for `SqlReader.from_connection()`.  It has the disadvantage that it's not really a factory method, because it always returns a `PrivateReader`, unlike the factory method on `SqlReader`.  And `PrivateReader` inherits from `SqlReader`, so we would be overriding `from_connection()` with a different method signature.  Regardless, we should change the `engine` parameter on `from_connection()` to be keyword-only.

We could just bite the bullet and overload the `PrivateReader` constructor even further.  This should be safe, since we can switch on the first positional argument `if not isinstance(reader, SqlReader): <call connection factory>`.  And the existing overloads are all intended to handle deprecation warnings (for: flipping order of first two positional arguments, renaming the epsilon argument, and deprecating options), which we ought to go ahead and deprecate.

One challenge with using the constructor is that the constructor allows positional arguments for all of the optional arguments.  

We could address this by switching the signature to use keyword-only arguments, but this would be a breaking change for anyone who depended on the positional syntax.

This is primarily problematic for the `delta` argument.  Having privacy parameters be positional is a problem, because we would like to be able to expand the privacy definition to include more features, such as `neighboring_definition` or `protect_floating_point`.  We want to be able to expand the privacy definition without breaking the API.  To accomplish this, we could use a pattern like:

```
conn = pyodbc.connect(dsn)
privacy = Privacy(epsilon=1.0, protect_floating_point=True)  # uses default delta
private_reader = PrivateReader(conn, privacy, meta)
```

This is definitely a breaking change, but might be worth it.  This moves metadata to the third positional argument, because metadata will eventually be optional.  If we are making a breaking change here, we could go ahead and make the metadata parameter keyword-only, so calling pattern would be:

```
conn = pyodbc.connect(dsn)
privacy = Privacy(epsilon=1.0, protect_floating_point=True)  # uses default delta
private_reader = PrivateReader(conn, privacy, meta=meta)
```

**Comment by joshua-oss** [comment]
Addressed in #386

**Comment by joshua-oss** [comment]
Added in #386
